src/ripple/ledger/detail/ReadViewFwdRange.ipp

```diff
@@ -52,6 +52,13 @@ ReadViewFwdRange<ValueType>::iterator::iterator(
 }
 
 template<class ValueType>
+std::unique_ptr<typename ReadViewFwdRange<ValueType>::iter_base>
+ReadViewFwdRange<ValueType>::iterator::extract()
+{
+    return std::move(impl_);
+}
```

Comment:
It is needed in order to use `std::upper_bound` in the definition of `RawStateTable::slesUpperBound` without changing the current `sles` API.  It converts a `sles::iterator` into a `std::unique_ptr<typename ReadViewFwdRange<ValueType>::iter_base>`.